src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -166,6 +166,33 @@ public Kernel() {
         context.put(SERVICE_TYPE_TO_CLASS_MAP_KEY, typeToClassMap);
     }
 
+    /**
+     * Construct the Kernel to be used for connectivity validation.
+     *
+     * @param currentConfig         current device configs
+     * @param deploymentConfig      Map of the configs being merged in from deployment
+     * @param deploymentTimestamp   timestamp of the deployment being merged in
+     */
+    public Kernel(DeviceConfiguration currentConfig, Map<String, Object> deploymentConfig,
```

Comment:
why are we creating a new kernel? This is concerning as there should only ever be 1 instance of the kernel and Context in a given process.

Reply:
I wasn't aware of this.

I created a new kernel because DeviceConfiguration uses it to get configuration values, and it didn't make sense to modify the configuration in the existing kernel

Reply:
Kernel doesn't hold the configurations does it. The config holds the configurations. So how about we find a way to remove the kernel from device config.

Reply:
In DeviceConfiguration I've added a new config variable and replaced all `kernel.getConfig()` calls with that variable.

Right now the kernel is still used for a few `getContext()` calls and one `getNucleusPaths()` and one `deTilde(Coerce.toString(newV))`

Should I try to refactor those so DeviceConfiguration no longer keeps track of kernel as a variable?

Reply:
Yes, the device configuration should not need the kernel. The initialization stuff is what seems to need context and kernel. You don't need any of that init stuff for your use case. See if you can extract that out to another class.